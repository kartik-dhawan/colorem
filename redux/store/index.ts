import { configureStore } from "@reduxjs/toolkit"
import countSlice from "../slices/countSlice"

const store = configureStore({
  reducer: {
    countSlice,
  },
})

export default store
