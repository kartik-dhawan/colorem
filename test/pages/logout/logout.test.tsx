import { Provider } from "react-redux"
import renderer from "react-test-renderer"
import "@testing-library/jest-dom"
import store from "../../../redux/store"
import { render } from "@testing-library/react"
import LogoutPage from "../../../pages/logout"

const logoutText = "Please wait while we log you out..."

jest.mock("next/router", () => require("next-router-mock"))

describe("Logout page", () => {
  it("-- should match snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <LogoutPage />
        </Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  it("-- should render logout text component correctly", () => {
    const view = render(
      <Provider store={store}>
        <LogoutPage />
      </Provider>
    )

    const logoutTextComponent = view.getByText(logoutText)
    expect(logoutTextComponent).toBeInTheDocument()
  })
})
