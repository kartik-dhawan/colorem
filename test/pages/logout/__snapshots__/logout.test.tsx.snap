// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout page -- should match snapshot 1`] = `
<div
  className="logoutPageWrapper MuiBox-root css-z7qmyv"
  data-testid="logoutPageWrapper"
  id="logoutPageWrapper"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1 logoutPageText css-122tm9-MuiTypography-root"
    data-testid="logoutPageText"
    id="logoutPageText"
  >
    Please wait while we log you out...
  </h6>
</div>
`;
